tools/plugin_sync/plugin_sync.pt

```diff
@@ -0,0 +1,200 @@
+  var github_raw_url = 'https://raw.githubusercontent.com/flexera/flexera-plugins/master/';
+
+  // create a map of published templates
+  for (var index = 0; index < published_plugin_list.length; index++) {
+    var published_plugin = published_plugin_list[index];
+    old_plugin_list[published_plugin.name] = published_plugin;
+  }
+
+  // loop through active plugins from s3 json file
+  for ( var i = 0; i < active_plugin_list.length; i++) {
+    var active_plugin = active_plugin_list[i]
+    // map catalog plugin with json list
+    var account_plugin = old_plugin_list[active_plugin.name]
+
+    var plugin_name = ""
+    var account_version = "0.0.0"
+    if (account_plugin){
+      var account_version = account_plugin.version
+      var plugin_name = account_plugin.name
+    }
+    // published version
+    var a_version = account_version.toString().split('.')
+    // repo version
+    var r_version = active_plugin.version.toString().split('.')
+
+    var version_upgradeable = false
+    curr_major_version = parseInt(a_version[0])
+    curr_minor_version = parseInt(a_version[1])
+    curr_patch_version = parseInt(a_version[2])
+    new_major_version = parseInt(r_version[0])
+    new_minor_version = parseInt(r_version[1])
+    new_patch_version =parseInt(r_version[2])
+
+    if (curr_major_version < new_major_version ) {
+      var version_upgradeable = true
+    } else if (curr_minor_version < new_minor_version) {
+        var version_upgradeable = true
+    } else if (curr_patch_version < new_patch_version) {
+        var version_upgradeable = true
+    }
+
+    if (force_upgrade == 1 ) {
+      var version_upgradeable = true
+    }
+
+    plugin_list.push({
+      name: active_plugin.name,
+      file_name: active_plugin.file_name,
+      repo_version: active_plugin.version,
+      account_version: account_version,
+      account_plugin: account_plugin,
+      version_upgradeable: version_upgradeable,
+      change_log: github_raw_url.concat(active_plugin.change_log)
+    })
+  }
+EOS
+end
+
+escalation "upload_new_plugins" do
+  run "upload_template", data, rs_project_id,rs_org_id
+end
+
+escalation "report_plugin_update" do
+  email $param_email
+end
+
+define upload_template($data,$project_id,$rs_org_id) return $url_array,$response_array do
+
+    $url_array = []
+    $response_array = []
+    foreach $item in $data do
+      $url = "https://raw.githubusercontent.com/flexera/flexera-plugins/master/"+to_s($item["file_name"])
+      $url_array << $url
+
+      $plugin_request = http_get(
+       insecure: false,
+       url: $url
+      )
+      $plugin_source = $plugin_request["body"]
+      $file_name = last(split($item["file_name"],'/'))
+
+      # add the new plugin
+      $create_response = http_request(
+        auth: $$auth_rs,
+        verb: "post",
+        https: true,
+        host: 'cloud-3.rightscale.com',
+        href: join(["/cloud/v1/orgs/",$rs_org_id,"/plugins"]),
+        headers: { "api-version": "1.0" },
```

Comment:
We do not need `api-version` header for the plugin api

Reply:
will remove